extensions/integration-tests/src/test/schemaCompare.test.ts

```diff
@@ -284,6 +268,23 @@ suite('Schema compare integration test suite', () => {
 	});
 });
 
+async function getConnectionUri(server: TestServerProfile): Promise<string> {
+	let result = await utils.retryFunction(async () => {
+		await utils.connectToServer(server, SERVER_CONNECTION_TIMEOUT);
```

Comment:
It feels like these should be in separate retry blocks - for example if the connection is successful then I don't think we need to retry making the connection over and over again.